tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/RecordEntry.cs

```diff
@@ -26,16 +26,54 @@ public class RecordEntry
 
         public RequestOrResponse Response { get; set; } = new RequestOrResponse();
 
-        public string RequestUri { get; set; }
+        private string _requestUri;
+        public string RequestUri
+        {
+            get { return this._requestUri; }
+            set {
```

Comment:
I think you went with the `InitializeIsModifiedFlag` because this `set` is causing the `isModified` to always be true. I'll help come up with a workaround.

Reply:
Leaving it as is, as discussed offline.